extension-localization/src/main/java/com/mapbox/maps/extension/localization/Localization.kt

```diff
@@ -29,13 +29,15 @@ internal fun setMapLanguage(locale: Locale, style: StyleInterface, layerIds: Lis
         }
         .forEach { layer ->
           val symbolLayer = style.getLayerAs<SymbolLayer>(layer.id)
-          symbolLayer.textFieldAsExpression?.let { textFieldExpression ->
-            if (BuildConfig.DEBUG) {
-              Logger.i(TAG, "Localize layer id: ${symbolLayer.layerId}")
+          symbolLayer?.let {
+            it.textFieldAsExpression?.let { textFieldExpression ->
+              if (BuildConfig.DEBUG) {
+                Logger.i(TAG, "Localize layer id: ${it.layerId}")
+              }
+              val language = if (sourceIsStreetsV8(style, source)) getLanguageNameV8(locale)
+              else getLanguageNameV7(locale)
+              convertExpression(language, it, textFieldExpression)
             }
```

Comment:
nit: can be simplified
```suggestion
          symbolLayer?.textFieldAsExpression?.let { textFieldExpression ->
              if (BuildConfig.DEBUG) {
                Logger.i(TAG, "Localize layer id: ${it.layerId}")
              }
              val language = if (sourceIsStreetsV8(style, source)) getLanguageNameV8(locale)
              else getLanguageNameV7(locale)
              convertExpression(language, it, textFieldExpression)
            }
```

Reply:
do you think this will work? in `convertExpression(language, it, textFieldExpression)` `it` is basically a `symbolLayer`, which will give an unresolved reference if we use it like this. let me check that too :D